wallet/wallet.go

```diff
@@ -2373,6 +2373,174 @@ func (w *Wallet) GetTransactions(startBlock, endBlock *BlockIdentifier,
 	return &res, err
 }
 
+// GetTransactionResult returns a transaction in the UnminedTransaction
+// field if it is unmined, or in the MinedTransaction field as a Block
+// structure for a mined transaction.
+type GetTransactionResult struct {
+	MinedTransaction   *Block
+	UnminedTransaction *TransactionSummary
+}
+
+// GetTransaction returns data of any transaction given its id. A mined
+// transaction is returned in a Block structure which records properties about
+// the block. A bool is also returned to denote if the transaction previously
+// existed in the database or not.
+// The backend must have a transaction index to poll a confirmed transaction
+// that does not exist in the internal wallet.
+func (w *Wallet) GetTransaction(txHash *chainhash.Hash) (*GetTransactionResult,
```

Comment:
Unclear w.r.t the purpose of this method: it seems to rely on the RPC calls, and will return a reply even if the wallet isn't found in the wallet. 

Where would this be used that https://pkg.go.dev/github.com/btcsuite/btcwallet/wtxmgr#Store.TxDetails wouldn't be?

Reply:
This could be useful for applications that use `btcwallet` as dependency. E.g. in LND a remotely force closed transaction is not added to the wallet/store and therefore won't show up via `GetTransactions`. It could therefore be convenient to have a function that gets the transaction from the wallet/store if it exists and falls back to the backend otherwise.